vllm/utils/flashinfer.py

```diff
@@ -357,7 +357,8 @@ def bmm_fp8(
     ) -> torch.Tensor:
         from flashinfer import bmm_fp8 as bmm_fp8_
 
-        return bmm_fp8_(A, B, A_scale, B_scale, dtype, None, backend)
+        result = bmm_fp8_(A, B, A_scale, B_scale, dtype, None, backend)
+        return result.reshape(A.shape[1], B.shape[2])
```

Comment:
I moved the reshape into the kernel because it was causing a lot of trouble in the pattern matching wrt dynamic shapes.. hope this is ok